#!/usr/bin/env python
PACKAGE = "tams_phasespace"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# name type level description default min max
gen.add( "workspace_dx", double_t, 0, "workspace x sample spacing", 0.1, 0.001, 1.0)
gen.add( "workspace_dy", double_t, 0, "workspace y sample spacing", 0.1, 0.001, 1.0)
gen.add( "workspace_dz", double_t, 0, "workspace z sample spacing", 0.1, 0.001, 1.0)
#
gen.add( "workspace_xmin", double_t, 0, "workspace lower x limit", -1.0, -100.0, 100.0)
gen.add( "workspace_xmax", double_t, 0, "workspace upper x limit", +1.0, -100.0, 100.0)
gen.add( "workspace_ymin", double_t, 0, "workspace lower y limit", -1.0, -100.0, 100.0)
gen.add( "workspace_ymax", double_t, 0, "workspace upper y limit", +1.0, -100.0, 100.0)
gen.add( "workspace_zmin", double_t, 0, "workspace lower z limit", -1.0, -100.0, 100.0)
gen.add( "workspace_zmax", double_t, 0, "workspace upper z limit", +1.0, -100.0, 100.0)
#
gen.add( "colormap",       int_t,    0, "colormap",   1, 0, 7 );
gen.add( "marker_size",    double_t, 0, "visualization marker size", 0.05, 0.001, 1.0)
gen.add( "enabled_cameras", str_t,   0, "enabled_cameras", "1,2,3,4,5,6,7,8")
gen.add( "parent_frame",   str_t,    0, "marker parent frame", "world")
gen.add( "debug_level",    int_t,    0, "debug_level",   1, 0, 5 );

exit(gen.generate(PACKAGE, "workspace_visualizer", "WorkspaceParameters"))
